### ISSUE TITLE
Crash on uncommitted files when `:Gitsigns preview_hunk` called

### ISSUE THREAD

**Issue by gytis-ivaskevicius** [lewis6991/gitsigns.nvim#554]
### Description

**Given:**
```
touch abc
git add abc
nvim abc
```

**When:**
`:gitsigns preview_hunk` is executed

**Then:**
An exception is thrown. error:
```
Error executing Lua callback: ...ack/nix2vim/start/gitsigns.nvim/lua/gitsigns/actions.lua:593: attempt to index local 'bcache' (a nil value)
stack traceback:
        ...ack/nix2vim/start/gitsigns.nvim/lua/gitsigns/actions.lua:593: in function 'f'
        ...ack/nix2vim/start/gitsigns.nvim/lua/gitsigns/actions.lua:582: in function <...ack/nix2vim/start/gitsigns.nvim/lua/gitsigns/actions.lua:579>
        ...ck-dir/pack/nix2vim/start/gitsigns.nvim/lua/gitsigns.lua:397: in function '_run_func'
        ...ck-dir/pack/nix2vim/start/gitsigns.nvim/lua/gitsigns.lua:480: in function <...ck-dir/pack/nix2vim/start/gitsigns.nvim/lua/gitsigns.lua:478>
        ```

### Neovim version

NVIM v0.7.0 Build type: Release LuaJIT 2.1.0-beta3 Compiled by nixbld  Features: +acl +iconv +tui See ":help feature-compile"     system vimrc file: "$VIM/sysinit.vim"   fall-back for $VIM: " /nix/store/8s7xlazpgfip0bfakcjvsa20zz11m0ga-neovim-unwrapped-0.7.0/share/nvim "  Run :checkhealth for more info

### Operating system and version

NixOS

### Expected behavior

No exception is thrown

### Actual behavior

Error

### Minimal config

```Lua
vim.o.packpath = '/tmp/nvim/site'

local plugins = {
  gitsigns = 'https://github.com/lewis6991/gitsigns.nvim',
  -- ADD OTHER PLUGINS _NECESSARY_ TO REPRODUCE THE ISSUE
}

for name, url in pairs(plugins) do
  local install_path = '/tmp/nvim/site/pack/test/start/'..name
  if vim.fn.isdirectory(install_path) == 0 then
    vim.fn.system { 'git', 'clone', '--depth=1', url, install_path }
  end
end

require('gitsigns').setup{
  debug_mode = true, -- You must add this to enable debug messages
  -- ADD GITSIGNS CONFIG THAT IS _NECESSARY_ FOR REPRODUCING THE ISSUE
}

-- ADD INIT.LUA SETTINGS THAT IS _NECESSARY_ FOR REPRODUCING THE ISSUE
```

### Steps to reproduce

**Given:**
```
touch abc
git add abc
nvim abc
```

**When:**
`:gitsigns preview_hunk` is executed

### Gitsigns debug messages

Error executing Lua callback: ...ack/nix2vim/start/gitsigns.nvim/lua/gitsigns/actions.lua:593: attempt to index local 'bcache' (a nil value)
stack traceback:
        ...ack/nix2vim/start/gitsigns.nvim/lua/gitsigns/actions.lua:593: in function 'f'
        ...ack/nix2vim/start/gitsigns.nvim/lua/gitsigns/actions.lua:582: in function <...ack/nix2vim/start/gitsigns.nvim/lua/gitsigns/actions.lua:579>
        ...ck-dir/pack/nix2vim/start/gitsigns.nvim/lua/gitsigns.lua:397: in function '_run_func'
        ...ck-dir/pack/nix2vim/start/gitsigns.nvim/lua/gitsigns.lua:480: in function <...ck-dir/pack/nix2vim/start/gitsigns.nvim/lua/gitsigns.lua:478>

**Comment by lewis6991** [comment]
Is your Gitsigns up to date? The line numbers don't look correct.

**Comment by lewis6991** [comment]
Fixed in 9b3100eaea174f5a321b728e6059fc29b2f58c9e
